packages/components/src/disabled/test/index.tsx

```diff
@@ -0,0 +1,178 @@
+ */
+import Disabled from '../';
+
+jest.mock( '@wordpress/dom', () => {
+	const focus = jest.requireActual( '../../../../dom/src' ).focus;
+	return {
+		focus: {
+			...focus,
+			focusable: {
+				...focus.focusable,
+				find( context: Element, options = { sequential: false } ) {
+					// In JSDOM, all elements have zero'd widths and height.
+					// This is a metric for focusable's `isVisible`, so find
+					// and apply an arbitrary non-zero width.
+					Array.from( context.querySelectorAll( '*' ) ).forEach(
+						( element ) => {
+							Object.defineProperties( element, {
+								offsetWidth: {
+									get: () => 1,
+									configurable: true,
+								},
+							} );
+						}
+					);
+
+					return focus.focusable.find( context, options );
+				},
+			},
+		},
+	};
+} );
+
+describe( 'Disabled', () => {
+	const Form = () => (
+		<form title="form">
+			<input />
+			<div title="edit my content" contentEditable tabIndex={ 0 } />
+		</form>
+	);
+
+	it( 'will disable all fields', () => {
+		render(
+			<Disabled>
+				<Form />
+			</Disabled>
+		);
+
+		const input = screen.getByRole( 'textbox' );
+		const contentEditable = screen.getByTitle( 'edit my content' );
+		expect( input ).toBeDisabled();
+		expect( contentEditable ).toHaveAttribute( 'contenteditable', 'false' );
+		expect( contentEditable ).not.toHaveAttribute( 'tabindex' );
+		expect( contentEditable ).not.toHaveAttribute( 'disabled' );
+	} );
+
+	it( 'should cleanly un-disable via reconciliation', () => {
+		const MaybeDisable = ( { isDisabled = true } ) =>
+			isDisabled ? (
+				<Disabled>
+					<Form />
+				</Disabled>
+			) : (
+				<Form />
+			);
+
+		const { rerender } = render( <MaybeDisable /> );
+
+		const input = screen.getByRole( 'textbox' );
+		const contentEditable = screen.getByTitle( 'edit my content' );
+
+		expect( input ).toBeDisabled();
+		expect( contentEditable ).toHaveAttribute( 'contenteditable', 'false' );
+
+		rerender( <MaybeDisable isDisabled={ false } /> );
+
+		expect( input ).not.toBeDisabled();
+		expect( contentEditable ).toHaveAttribute( 'contenteditable', 'true' );
+		expect( contentEditable ).toHaveAttribute( 'tabindex' );
+	} );
+
+	it( 'will disable or enable descendant fields based on the isDisabled prop value', () => {
+		const MaybeDisable = ( { isDisabled = true } ) => (
+			<Disabled isDisabled={ isDisabled }>
+				<Form />
+			</Disabled>
+		);
+
+		const { rerender } = render( <MaybeDisable /> );
+
+		const input = screen.getByRole( 'textbox' );
+		const contentEditable = screen.getByTitle( 'edit my content' );
+
+		expect( input ).toBeDisabled();
+		expect( contentEditable ).toHaveAttribute( 'contenteditable', 'false' );
+
+		rerender( <MaybeDisable isDisabled={ false } /> );
+
+		expect( input ).not.toBeDisabled();
+		expect( contentEditable ).toHaveAttribute( 'contenteditable', 'true' );
+	} );
+
+	// Ideally, we'd have two more test cases here:
+	//
+	//  - it( 'will disable all fields on component render change' )
+	//
```

Comment:
I'm not really sure that the original author meant with this suggestion — we can probably remove this comment

Reply:
I didn't really know what to write in this test.  🥴 Thanks!